[Access for SaaS] Grafana and Grafana Cloud tutorial (#15649)
cloudflare/cloudflare-docs: content/cloudflare-one/applications/configure-apps/saas-apps/grafana-saas-oidc.md

---
pcx_content_type: how-to
title: Grafana
weight: 10
---

# Connect to Grafana through Access

This guide covers how to configure [Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/) as an OIDC application in Cloudflare Zero Trust.

## Prerequisites

- An [OIDC identity provider](/cloudflare-one/identity/idp-integration/generic-oidc/) configured in Cloudflare Zero Trust
- Admin access to a Grafana account

{{<Aside type="note">}}

> You can also configure OIDC SSO for Grafana using the [configuration file](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/#configure-generic-oauth-authentication-client-using-the-grafana-configuration-file), instead of using the user interface (UI) that is documented here.

Reviewer: ```suggestion
You can also configure OIDC SSO for Grafana using a [configuration file](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/generic-oauth/#configure-generic-oauth-authentication-client-using-the-grafana-configuration-file) instead of using Grafana's user interface (UI), as documented in this guide.
```

{{</Aside>}}

## 1. Add a SaaS application to Cloudflare Zero Trust

1. In [Zero Trust](https://one.dash.cloudflare.com), go to **Access** > **Applications**.
2. Select **SaaS**.
3. For **Application**, select _Grafana_.
4. For the authentication protocol, select **OIDC**.
5. Select **Add application**.
6. In **Scopes**, select the attributes that you want Access to send in the ID token.
7. In **Redirect URLs**, enter `https://<your-grafana-domain>/login/generic_oauth`.
8. (Optional) Enable [Proof of Key Exchange (PKCE)](https://www.oauth.com/oauth2-servers/pkce/) if the protocol is supported by your IdP. PKCE will be performed on all login attempts.
9. Copy the **Client secret**, **Client ID**, **Token endpoint**, and **Authorization endpoint**.
10. Select **Save configuration**.
11. (Optional) configure [App Launcher settings](/cloudflare-one/applications/app-launcher/) by turning on **Enable App in App Launcher** and, in **App Launcher URL**, entering `https://<your-grafana-domain>/login`.
12. Configure [Access policies](/cloudflare-one/policies/access/) for the application.
13. Select **Done**.

> ## 2. Add a SAML SSO provider to Grafana

Reviewer: ```suggestion
## 2. Add a SSO provider to Grafana
```
